src/sql/src/plan/query.rs

```diff
@@ -3019,7 +3020,16 @@ fn plan_subscript(
     let expr = plan_expr(ecx, expr)?.type_as_any(ecx)?;
     let ty = ecx.scalar_type(&expr);
     match &ty {
-        ScalarType::Array(..) => plan_subscript_array(ecx, expr, positions),
+        ScalarType::Array(..) | ScalarType::Int2Vector => plan_subscript_array(
```

Comment:
Not directly related to this PR, but I think the comment on this function is no longer valid?
> // Currently, slicing subscripts only work on `list`s, but should be expanded
// to arrays.

Reply:
Ah, this is confusing. The comment is true but its location makes it feel non-sequitur. There was already a comment to express this limitation in a spot more germane, so removing this.